core/time/src/clock.rs

```diff
@@ -202,12 +245,16 @@ impl FakeClock {
             if inner.now() >= t {
                 return;
             }
+
             let (sender, receiver) = tokio::sync::oneshot::channel();
             let waiter = ClockWaiterInHeap { waker: sender, deadline: t };
             inner.waiters.push(waiter);
             receiver
         };
-        receiver.await.unwrap();
+        
+        if receiver.await.is_err() {
```

Comment:
If this fails, we will continue operating.  This makes me nervous.  If this failed, then something unexpected has gone wrong.  Unless we have a proper understanding of the scenarios where this can go wrong and test cases covering it, I think it is better to crash.  I think we should keep the `unwrap`.